server/paymentProviders/opencollective/virtualcard.js

```diff
@@ -109,10 +109,16 @@ async function processOrder(order) {
   const sourcePaymentMethodProvider = libpayments.findPaymentMethodProvider(sourcePaymentMethod);
 
   // gets the Credit transaction generated
-  let creditTransaction = await sourcePaymentMethodProvider.processOrder(order);
-  // undo modification of original order after processing the source payment method order
-  order.PaymentMethodId = paymentMethod.id;
-  order.paymentMethod = paymentMethod;
+  let creditTransaction = null;
+  try {
+    creditTransaction = await sourcePaymentMethodProvider.processOrder(order);
+  } finally {
+    // undo modification of original order after processing the source payment method order
+    await order.update({ PaymentMethodId: paymentMethod.id });
```

Comment:
Can you point on the operation that we're undoing?

Reply:
This one: https://github.com/opencollective/opencollective-api/blob/65bbb86d4c8c2fc25609c9058f2f6582aa09d62b/server/paymentProviders/opencollective/virtualcard.js#L106

Reply:
Thank you. Never really looked at that code, sad that we're doing that :-/

Reply:
Agree